### PR TITLE
Added tests about Context>>home and Context>>activeHome

### PR DESCRIPTION
**Pull request by dupriezt**
Solves #2277

### CODE REVIEW

**Comment by MarcusDenker** [comment]
can we merge ContextTest2 with ContextTest?

**Comment by dupriezt** [comment]
It's possible. It would be inlining the setUp from ContextTest2 into the two test methods that use it.
Do you want me to do it?

**Comment by MarcusDenker** [comment]
yes!

**Comment by dupriezt** [comment]
Done

**Comment by MarcusDenker** [comment]
error not related

**Comment by Ducasse** [comment]
This is super cool cool to get more tests and knowledge on that part of the system.
Thanks a lot.

File: src/Kernel-Tests/ContextTest.class.st [hunk 1]
```diff
@@ -100,6 +113,17 @@ ContextTest >> testActivateReturnValue [
 	self assert:  ((aSender activateReturn: aMethodContext value: #()) receiver = aMethodContext).
 ]
 
+{ #category : #tests }
+ContextTest >> testActiveBlockContextHomeAndActiveHome [
+	[ self assert: thisContext home == thisContext activeHome.] value.
```
Comment (astares) [inline src/Kernel-Tests/ContextTest.class.st:118]: no dot necessary in the block

File: src/Kernel-Tests/ContextTest.class.st [hunk 2]
```diff
@@ -34,6 +36,17 @@ ContextTest class >> contextWithTempForTesting [
...
 
+{ #category : #helper }
+ContextTest class >> createBlock [
+	^ [ thisContext ].
+]
+
+{ #category : #helper }
+ContextTest class >> returnNonActiveContextOfBlock [
+	Block := self createBlock.
+	^ Block value.
```
Comment (astares) [inline src/Kernel-Tests/ContextTest.class.st:47]: No dot necessary at end of  method

File: src/Kernel-Tests/ContextTest.class.st [hunk 3]
```diff
@@ -100,6 +113,17 @@ ContextTest >> testActivateReturnValue [
...
 ]
 
+{ #category : #tests }
+ContextTest >> testActiveBlockContextHomeAndActiveHome [
+	[ self assert: thisContext home == thisContext activeHome.] value.
+]
+
+{ #category : #private }
+ContextTest >> testActiveHomeMethodContext [
+	self assert: (aMethodContext activeHome == aMethodContext).
```
Comment (astares) [inline src/Kernel-Tests/ContextTest.class.st:123]: could be chained as

```Smalltalk
self 
   assert: (aMethodContext activeHome == aMethodContext);
   assert: (aMethodContext activeHome == aMethodContext home)
```

File: src/Kernel-Tests/ContextTest.class.st [hunk 4]
```diff
@@ -120,6 +144,18 @@ ContextTest >> testMethodIsBottomContext [
 	self assert: aMethodContext secondFromBottom = aMethodContext.
```
Comment (astares) [inline src/Kernel-Tests/ContextTest.class.st:144]: no dot necessary at the end

**Comment by dupriezt** [comment]
Ok, but the pull request is already merged and I'm not sure it's worth creating another PR to remove dots at the end of methods.
Or is it?

**Comment by guillep** [comment]
It is :) Read about the broken window theory :)

**Comment by astares** [comment]
Yes, please provide another new issue with pull request. We want get and stay cleaner and cleaner.

In a nice environment you will write nice code - in a "dirty" environment people will follow bad practices...

**Comment by jecisc** [comment]
I think you car still commit on this branch and re-open a PR from it.

**Comment by estebanlm** [comment]
yes, you can always do that

File: src/Kernel-Tests/ContextTest.class.st (hunk 3 above)
Comment (estebanlm) [reply src/Kernel-Tests/ContextTest.class.st:123]: and honestly, I would do better: 

```Smalltalk
self 
   assert: aMethodContext activeHome equals: aMethodContext;
   assert: aMethodContext activeHome equals: aMethodContext home
```

**Comment by dupriezt** [comment]
Done: #2315

File: src/Kernel-Tests/ContextTest.class.st (hunk 3 above)
Comment (dupriezt) [reply src/Kernel-Tests/ContextTest.class.st:123]: `assert: ==` translates to `assert: identicalTo:`, not `assert: equal:`, right?

**Pull request merged**